frontend/src/components/container/sidenav/Sidenav.tsx

```diff
@@ -1,13 +1,19 @@
 import { Dispatch } from 'react';
 import { Link } from 'react-router-dom';
-import { Home, Widgets, CalendarToday } from '@material-ui/icons';
+import { Home, Widgets, SupervisorAccount, CalendarToday } from '@material-ui/icons';
 
 import styles from './Sidenav.module.css';
+import jwtDecode from 'jwt-decode';
 
 interface Props {
   showSidenav: boolean;
   toggleSidenav: Dispatch<boolean>;
 }
+let userTokenDecoded: any = null;
+
+if (localStorage.getItem('token') != null || localStorage.getItem('token') != undefined) {
+  userTokenDecoded = jwtDecode(localStorage.getItem('token') as string);
```

Comment:
Why do you need to decode the token?

Reply:
To get the user role to conditionally render the admin tab in the side bar

Reply:
Ah ok

Reply:
To get the user role.

Reply:
Only an Admin user should have access to the Admin page.